plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -303,20 +307,20 @@ struct block_time_tracker {
       assert(!paused);
       if( _log.is_enabled( fc::log_level::debug ) ) {
          auto diff = now - clear_time_point - block_idle_time - trx_success_time - trx_fail_time - transient_trx_time - other_time;
-         fc_dlog( _log, "Block #${n} ${p} trx idle: ${i}us out of ${t}us, success: ${sn}, ${s}us, fail: ${fn}, ${f}us, "
+         fc_dlog( _log, "Block #${n} ${p} trx idle: ${i}us out of ${t}us, success: ${sn}, ${s}us, exhausted: ${en}, ${e}us, fail: ${fn}, ${f}us, "
                   "transient: ${ttn}, ${tt}us, other: ${o}us${rest}",
                   ("n", block_num)("p", producer)
                   ("i", block_idle_time)("t", now - clear_time_point)("sn", trx_success_num)("s", trx_success_time)
-                  ("fn", trx_fail_num)("f", trx_fail_time)
+                  ("en", trx_exhausted_num)("e", trx_exhausted_time)("fn", trx_fail_num)("f", trx_fail_time)
                   ("ttn", transient_trx_num)("tt", transient_trx_time)
                   ("o", other_time)("rest", diff.count() > 5 ? ", diff: "s + std::to_string(diff.count()) + "us"s : ""s ) );
       }
    }
 
    void clear() {
       assert(!paused);
-      block_idle_time = trx_fail_time = trx_success_time = transient_trx_time = other_time = fc::microseconds{};
-      trx_fail_num = trx_success_num = transient_trx_num = 0;
+      block_idle_time = trx_success_time = trx_exhausted_time = trx_fail_time = transient_trx_time = other_time = fc::microseconds{};
+      trx_success_num = trx_exhausted_num = trx_fail_num = transient_trx_num = 0;
```

Comment:
nit: not requesting a change, just thinking that, at this point, I think it would be cleaner to have an array of `struct trx_result { fc::microseconds time_spent, uint32_t num_trx };` indexed by `time_status`.